### PR TITLE
Fix RecordSensor serialization

### PR DESCRIPTION
**Pull request by chetan51**
Fixes https://github.com/numenta/nupic/issues/2922.

### CODE REVIEW

**Comment by chetan51** [comment]
@scottpurdy

File: src/nupic/regions/RecordSensor.py [hunk 1]
```diff
@@ -211,7 +211,7 @@ def __init__(self, verbosity=0, numCategories=1):
 
     self.preEncodingFilters = []
     self.postEncodingFilters = []
-    self.topDownMode = False
+    self.topDownMode = 0
```
Comment (scottpurdy) [inline src/nupic/regions/RecordSensor.py]: Do you have to change this? A boolean makes more sense to me.

**Comment by scottpurdy** [comment]
:+1:

**Comment by chetan51** [comment]
@scottpurdy Update to incorporate your feedbakc.

File: src/nupic/regions/RecordSensor.py [hunk 2]
```diff
@@ -548,8 +549,9 @@ def readFromProto(cls, proto):
     instance = cls()
 
     instance.encoder = MultiEncoder.read(proto.encoder)
-    instance.disabledEncoder = MultiEncoder.read(proto.disabledEncoder)
-    instance.topDownMode = proto.topDownMode
+    if proto.disabledEncoder is not None:
+      instance.disabledEncoder = MultiEncoder.read(proto.disabledEncoder)
+    instance.topDownMode = True if proto.topDownMode else False
```
Comment (scottpurdy) [inline src/nupic/regions/RecordSensor.py]: I believe bool(proto.topDownMode) will work here. And above you may be able to do int(self.topDownMode)

**Comment by chetan51** [comment]
@scottpurdy Updated.

**Pull request merged**